tests/Feature/Http/Controllers/Api/PostControllerTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Feature;
+
+use App\Models\Post;
+use App\Models\User;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Laravel\Sanctum\Sanctum;
+
+use function Pest\Laravel\assertDatabaseHas;
+use function Pest\Laravel\deleteJson;
+use function Pest\Laravel\getJson;
+use function Pest\Laravel\postJson;
+use function Pest\Laravel\putJson;
+
+uses(RefreshDatabase::class);
+
+beforeEach(function () {
+    Sanctum::actingAs(
+        User::factory()->create(),
+        ['*']
+    );
+});
+
+it('can list all posts', function () {
+    $posts = Post::factory()->count(3)->create();
+
+    getJson('/api/posts')
+        ->assertOk()
+        ->assertJson($posts->toArray());
+});
+
+it('can create a new post', function () {
+    $data = Post::factory()->make()->toArray();
```

Comment:
It's not necessary to convert this to an array.

Reply:
not working without it